### PR TITLE
Apply some normalization on Gitlab API handling

### PR DESCRIPTION
**Pull request by shulard**
I've found that there are simpler methods to retrieve data from Gitlab API library than the one used for the moment.

Also, Gitlab API response structure is not the same than Github, some keys must be updated.

### CODE REVIEW

File: src/ThirdParty/Gitlab/Adapter/GitLabRepoAdapter.php [hunk 1]
```diff
@@ -80,13 +78,16 @@ public function createComment($id, $message)
...
     {
-        $issue = MergeRequest::fromArray(
-            $this->client,
-            $this->getCurrentProject(),
-            $this->client->api('merge_requests')->show($this->getCurrentProject()->id, $id)
-        );
-
-        return $issue->showComments();
+        $comments = $this->client->api('merge_requests')->showNotes($this->getCurrentProject()->id, $id);
+        return array_filter(array_map(function($note) {
```
Comment (sstok) [inline src/ThirdParty/Gitlab/Adapter/GitLabRepoAdapter.php]: Please run the php-cs-fixer 👍

**Comment by shulard** [comment]
Ok, I've ran php-cs-fixer...

File: src/Command/Issue/IssueShowCommand.php [hunk 2]
```diff
@@ -89,20 +84,21 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-                        $comment['created_at']->format('r')
-                    ),
-                    'Link: '.$comment['url'],
-                    '',
-                    wordwrap($comment['body'], 100),
+                $styleHelper->text(sprintf(
+                    '<fg=white>Comment #%s</> by %s on %s',
+                    $comment['id'],
+                    $comment['user'],
+                    empty($comment['created_at']) ? '' : $comment['created_at']->format('r')
```
Comment (cordoval) [inline src/Command/Issue/IssueShowCommand.php]: do you have to do this check?

File: src/Command/Issue/IssueShowCommand.php [hunk 3]
```diff
@@ -89,20 +84,21 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+                ));
+                $styleHelper->detailsTable([
```
Comment (cordoval) [inline src/Command/Issue/IssueShowCommand.php]: inline this and the next line

File: src/Command/PullRequest/PullRequestListCommand.php [hunk 4]
```diff
@@ -47,7 +47,7 @@ protected function configure()
                 <<<EOF
 The <info>%command.name%</info> command lists all the pull requests:
 
-    <info>$ gush %command.name%</info>
+    <info>$ gush %command.name% 12</info>
```
Comment (cordoval) [inline src/Command/PullRequest/PullRequestListCommand.php]: what is the state 12?

File: src/ThirdParty/Gitlab/Adapter/GitLabIssueTracker.php [hunk 5]
```diff
@@ -188,29 +188,23 @@ public function createComment($id, $message)
      */
     public function getComments($id)
     {
-        $issue = Issue::fromArray(
```
Comment (cordoval) [inline src/ThirdParty/Gitlab/Adapter/GitLabIssueTracker.php]: check if fromArray is still used, perhaps it needs to be removed?

**Comment by cordoval** [comment]
- [ ] add todo list of what is left to the PR description
- [ ] manually test it

File: src/Command/Issue/IssueShowCommand.php (hunk 2 above)
Comment (shulard) [inline src/Command/Issue/IssueShowCommand.php]: The `created_at` key can be null if there is no date in API data... Maybe it's too much, there is always a date... I'll remove it !

File: src/Command/PullRequest/PullRequestListCommand.php (hunk 4 above)
Comment (shulard) [inline src/Command/PullRequest/PullRequestListCommand.php]: Hum I don't know... I'll remove it

File: src/ThirdParty/Gitlab/Adapter/GitLabIssueTracker.php (hunk 5 above)
Comment (shulard) [inline src/ThirdParty/Gitlab/Adapter/GitLabIssueTracker.php]: Ok I'll take a look at that.

**Comment by shulard** [comment]
Hello @cordoval you were right 😄. There are other places where API calls can be optimized. I've commited the changes. I've also processed your different feedbacks.

**Comment by cordoval** [comment]
@shulard the build will push the new images on dockerhub for gush, after it builds and pushes successfully, can you docker pull and test these changes please? thanks so we have some validation. Thanks

**Comment by cordoval** [comment]
test with github and gitlab

**Comment by shulard** [comment]
ok I'll try with docker 😄

**Comment by shulard** [comment]
Hello @cordoval 

Is the image already built ? I've pull from docker hub and I've got some errors against data manipulation (array to string, strstr with a first parameter as array...). On my local machine without docker, there is no problem with data structure...

**Comment by cordoval** [comment]
paste the errors and how you are running it, the image was built https://hub.docker.com/r/coder20078/gush/tags/ 20 hours ago, make sure you do docker pull coder20078/gush again, check how old is your images with docker images

**Comment by shulard** [comment]
I've just made `docker pull coder20078/gush` five minute ago.

```
$ docker images
REPOSITORY                                    TAG                 IMAGE ID            CREATED             SIZE
coder20078/gush                               latest              5cea92fb2356        19 hours ago        112.2 MB
```

I ran it using the `gush.dist` script bundled in the repo.

Example here :

```
$ ~/Sites/Forks/Gush/gush.dist issue:show 320186

Issue #320186 - Add a valid unit test suite !!! by shulard [opened]
===================================================================


Warning: strstr() expects parameter 1 to be string, array given in /usr/src/gush/vendor/symfony/console/Helper/Table.php on line 387

Notice: Array to string conversion in /usr/src/gush/vendor/symfony/console/Formatter/OutputFormatter.php on line 139
 Org/Repo:  bee4/beebot
 Link:      https://gitlab.com/bee4/beebot/issues/31
 Labels:    Bot, confirmed, critical

Warning: mb_detect_encoding() expects parameter 1 to be string, array given in /usr/src/gush/vendor/symfony/console/Helper/Table.php on line 342

Warning: strlen() expects parameter 1 to be string, array given in /usr/src/gush/vendor/symfony/console/Helper/Table.php on line 343

Warning: mb_strwidth() expects parameter 1 to be string, array given in /usr/src/gush/vendor/symfony/console/Helper/Table.php on line 343

Warning: mb_detect_encoding() expects parameter 1 to be string, array given in /usr/src/gush/vendor/symfony/console/Helper/Helper.php on line 54

Warning: mb_strwidth() expects parameter 1 to be string, array given in /usr/src/gush/vendor/symfony/console/Helper/Helper.php on line 58

Notice: Array to string conversion in /usr/src/gush/vendor/symfony/console/Formatter/OutputFormatter.php on line 139

Notice: Array to string conversion in /usr/src/gush/vendor/symfony/console/Helper/Table.php on line 353
 Milestone: Array
 Assignee:  shulard

Body
----

 It is really important to be able to validate the bot behaviour
```

I'm using the master branch freshly updated.

**Comment by cordoval** [comment]
it works, the only problem are those warnings, which means that we are plugging an array instead somewhere.

**Comment by shulard** [comment]
Yeah I know but with the same code base I don't understand why these warning are present on the docker version...

**Comment by cordoval** [comment]
because your error reporting is set to ignore these, also because really you don't have the same environment.

**Pull request merged**